Fix MethodEnter support for Perf Agent

mpirvu opened the pull request:
In order to enable method tracing we need to set the
`can_generate_method_entry_events` capability. For some
reason this capability can be enabled only during the
OnLoad phase. Therefore, the agent must be loaded from
the very beginning, and late attach will not do.

This commit also adds the ability to print the stack trace
up to the specified depth.

Examples of commands:
```
  {
    "functionality": "methodEntryEvents",
    "command": "start",
    "delay": 1,
    "stackTraceDepth": 5
  },
  {
    "functionality": "methodEntryEvents",
    "command": "stop",
    "delay": 5
  }
```

Signed-off-by: Marius Pirvu <mpirvu@ca.ibm.com>

mpirvu commented:
@dsouzai Could you please review this PR? Thanks

dsouzai commented on perf-tool/src/agentOptions.cpp:
```diff
@@ -175,34 +180,23 @@ void modifyObjectAllocEvents(const std::string& function,const std::string& comm
     }
 }
 
-void modifyMonitorStackTrace(const std::string& function, const std::string& command)
-{
-    /* enable stack trace */
-    if (!command.compare("start"))
-    {
-        setMonitorStackTrace(true);
-    }
-    else if (!command.compare("stop"))
-    {
-        setMonitorStackTrace(false);
-    }
-    else
-    {
-        invalidCommand(function, command);
-    }
-}
```
This seems to be unrelated to the method enter/exit events; is this deleting the ability for the user to send `monitorStackTrace`?

mpirvu replied:
Yes. The stacktrace option is sent with the "start" command.
The exiting code actually did nothing

dsouzai commented on perf-tool/src/methodEntry.cpp:
```diff
@@ -33,65 +33,76 @@
+*/
+        sendToServer(j.dump());
     }
-    mEntrySampleCount = atomic_fetch_add(&mEntrySampleCount, 1);
```
Don't we need this? Otherwise `mEntrySampleCount` will never get incremented.

mpirvu replied:
There is another increment on line 45. This one was a duplicate.

dsouzai reviewed (changes requested):
Should also remove the `void setMethodEntrySampleRate(int rate);` declaration in `methodEntry.hpp`.

dsouzai commented:
@mpirvu just waiting for the declaration of `setMethodEntrySampleRate` to be removed from `methodEntry.hpp` before I merge.

mpirvu commented:
@dsouzai I deleted the stale declaration of `setMethodEntrySampleRate` and corrected a couple of places where `sendToServer` is called to add a second parameter (event type), which is needed after #37 was merged.
